### DISCUSSION TITLE
Reusing getStaticProps method

### DISCUSSION THREAD

**Discussion by rinkusam12** [vercel/next.js#17257] [Help] [answered]
Hello There,
 Anybody have any example for reusing getStaticProps method. What I mean is I have few pages where I am getting content from the CMS in the getStaticProps. Now I am also fetching the content of header and footer from the CMS and writing same thing again and again seems kind of a boilerplate to me. As of right now, the _app.js don't support the getStaticProps method so I am looking for a way to reusing a getStaticProps method and extend it whenever I want to fetch more.

**Comment by lfades** [comment, marked as answer]
Not recommended at all. What you can do is write a function that returns what you need for the page and then use it in all of your pages. Eg:

```js
// lib/data.js
export default async function getData() {
  return { ... }
}

// pages/my-page.js
import getData from '../lib/data'

// Repeat this for other pages with the same requirements
export function getStaticProps() {
  return { props: await getData() }
}
```

It's important to have it that way so Next.js can know properly what code to remove from the browser build.

**Reply by romellem** [reply]
If this isn't recommended, what are the downsides to this? Would it be possible to provide a simple example showing that with this approach, your build ends up being larger than repeating the data fetching?

**Reply by madeleineostoja** [reply]
From my testing doing this breaks dead code elimination for any imports used in the `getData` method, which is especially problematic (and just caught me out) when using serverside only deps

**Reply by michael-odonovan** [reply]
Really sorry but I am having trouble implementing this.
What exactly should be inside the getData return { ???? }
thanks for you knowledge help

**Comment by Dupflo** [comment]
Yes the fact is you can use the old getInitialProps method in your _app to fetch your header and footer from the CMS API and then provide the props to you components also available in the _app.  However the downside is that it will remove your static site generation and the getStaticProps of your other pages on the build. So, I will effectively consider @lfades solution's until Next team does something about it

**Comment by cristianofmc** [comment]
does anyone have any updates on this? I'm having the same problem, I can't reuse components that need getStaticProps
